core/base/src/TError.cxx

```diff
@@ -131,17 +129,12 @@ void ErrorHandler(Int_t level, const char *location, const char *fmt, va_list ap
       fmt = "no error message provided";
 
    Int_t n = vsnprintf(buf, buf_size, fmt, ap);
-   // old vsnprintf's return -1 if string is truncated new ones return
-   // total number of characters that would have been written
-   if (n == -1 || n >= buf_size) {
-      if (n == -1)
-         buf_size *= 2;
-      else
-         buf_size = n+1;
+   if (n >= buf_size) {
```

Comment:
Do you know when (OS and/or compiler) the "old vsnprintf" behavior stopped being implemented?

Reply:
Some googling reveals the following:

> Some of the first implementations of sprintf() (before the
> C99 standard, e.g. the GNU libc 5) did return -1 if the
> resulting string did not fit into memory allowed for writing.

Apparently the same for old MS compilers, e.g. VC++ 6.0.  

In any case, the new behavior was standardized with C99 and C++11, so I guess we should be good.

Reply:
Make sure this info is in the commit message.